spring-cloud-azure-keyvault-config/src/main/java/com/microsoft/azure/spring/cloud/keyvault/config/KeyVaultConfigBootstrapConfiguration.java

```diff
@@ -6,31 +6,43 @@
 
 package com.microsoft.azure.spring.cloud.keyvault.config;
 
-import com.microsoft.azure.keyvault.KeyVaultClient;
-import com.microsoft.azure.spring.cloud.autoconfigure.telemetry.TelemetryAutoConfiguration;
-import com.microsoft.azure.spring.cloud.autoconfigure.telemetry.TelemetryCollector;
-import com.microsoft.azure.spring.cloud.keyvault.config.auth.*;
-import org.springframework.boot.autoconfigure.AutoConfigureBefore;
+import javax.annotation.PostConstruct;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.autoconfigure.AutoConfigureAfter;
 import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
 import org.springframework.boot.context.properties.EnableConfigurationProperties;
 import org.springframework.context.annotation.Bean;
 import org.springframework.context.annotation.Configuration;
 
-import javax.annotation.PostConstruct;
+import com.microsoft.azure.keyvault.KeyVaultClient;
+import com.microsoft.azure.spring.cloud.autoconfigure.telemetry.TelemetryAutoConfiguration;
+import com.microsoft.azure.spring.cloud.keyvault.config.auth.AadKeyVaultCredentials;
+import com.microsoft.azure.spring.cloud.keyvault.config.auth.AuthenticationExecutor;
+import com.microsoft.azure.spring.cloud.keyvault.config.auth.AuthenticationExecutorFactory;
+import com.microsoft.azure.spring.cloud.keyvault.config.auth.Credentials;
+import com.microsoft.azure.spring.cloud.keyvault.config.auth.DefaultAuthenticationExecutorFactory;
+import com.microsoft.azure.spring.cloud.telemetry.TelemetryCollector;
 
 /**
- * Spring Cloud Bootstrap Configuration for setting up an {@link KeyVaultPropertySourceLocator}.
+ * Spring Cloud Bootstrap Configuration for setting up an
+ * {@link KeyVaultPropertySourceLocator}.
  */
 @Configuration
-@AutoConfigureBefore(TelemetryAutoConfiguration.class)
 @ConditionalOnProperty(name = KeyVaultConfigProperties.ENABLED, matchIfMissing = true)
 @EnableConfigurationProperties(KeyVaultConfigProperties.class)
+@AutoConfigureAfter(TelemetryAutoConfiguration.class)

```

Comment:
Since the initial delay in the sender is 0, it will only be sent in the next message, an hour later.
I would suggest adding an initial delay to `TelemetrySender::sendEvent`. We need to get away from having global state in a singleton (in part because of the heinous static dependencies it introduces) and instead delegate the management of `TelemetryCollector` to Spring Configuration. In this scheme, all consumers of the Telemetry package will have to be autoconfigured after telemetry is configured.

Reply:
Got it.